internal/handlers/handlers.go

```diff
@@ -0,0 +1,40 @@
+package handlers
+
+import (
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+
+	"github.com/darod1n/urlshorten/internal/helpers"
+)
+
+type Storage interface {
+	AddURL(url string, shortURL string)
+	GetURL(shortURL string) (string, bool)
+}
+
+func APIShortURL(serverHost string, db Storage, res http.ResponseWriter, req *http.Request) {
+	if body, err := io.ReadAll(req.Body); err != nil {
+		res.WriteHeader(http.StatusBadRequest)
+	} else {
+		shortURL := helpers.GenerateShortURL(6)
+		db.AddURL(string(body), shortURL)
+		res.WriteHeader(http.StatusCreated)
+		resultURL, _ := url.JoinPath(serverHost, shortURL)
+		_, err := res.Write([]byte(resultURL))
+		if err != nil {
+			log.Fatalf(err.Error())
```

Comment:
Не надо вызывать `os.Exit` из хендлера - пока можно просто залоигровать ошибку

Кроме того:
```suggestion
			log.Fatal(err)
```